### PR TITLE
Support for RP initiated logout

### PR DESCRIPTION
**Pull request by rohe**
- [ ] Any changes relevant to users are recorded in the `CHANGELOG.md`.
- [ ] The documentation has been updated, if necessary.
---

### CODE REVIEW

File: src/oic/oic/provider.py [hunk 1]
```diff
@@ -553,47 +544,118 @@ def end_session_endpoint(self, request="", cookie=None, **kwargs):
...
+                if not match:
+                    return error_response('invalid_request', "Wrong user")
+
+            if not client_id:
+                if len(id_token_hint['aud']) == 1:
+                    client_id = id_token_hint['aud'][0]
+                else:
+                    client_id = id_token_hint['azp']
+
+            if not sids:
```
Comment (tpazderka) [inline src/oic/oic/provider.py]: This can be handled in `else` branch of `if` on line 585.

File: src/oic/oic/provider.py [hunk 2]
```diff
@@ -553,47 +544,118 @@ def end_session_endpoint(self, request="", cookie=None, **kwargs):
...
-        if "post_logout_redirect_uri" in esr:
-            redirect_uri = self.verify_post_logout_redirect_uri(esr, cookie)
-            if not redirect_uri:
-                msg = "Post logout redirect URI verification failed!"
-                return error_response("%s", msg)
+        # 2 ways of find out client ID and user. Either through a cookie
+        # or using the id_token_hint
+        client_id = uid = None
+        sids = []
+        cd = None
```
Comment (tpazderka) [inline src/oic/oic/provider.py]: `cookie_dealer` is probably better as a variable name.

File: src/oic/oauth2/provider.py [hunk 3]
```diff
@@ -728,33 +729,35 @@ def _complete_authz(self, user, areq, sid, **kwargs):
...
-                        _c = SimpleCookie()
-                        _c[name] = val
-                        _x = _c.output()
-                        if PY2:
-                            _x = str(_x)
-                        headers.append(tuple(_x.split(": ", 1)))
-                else:
+            _kaka = None
+
+        c_val = "{}][{}".format(user, areq['client_id'])
```
[the remark this answers is not in the document]
Comment (rohe) [reply src/oic/oauth2/provider.py]: I agree. I was on the way but something else intervened.

File: src/oic/oic/provider.py [hunk 4]
```diff
@@ -1,15 +1,7 @@
 #!/usr/bin/env python
-from future.backports.urllib.parse import splitquery
```
[the remark this answers is not in the document]
Comment (rohe) [reply src/oic/oic/provider.py]: ? Not sure I understand what you mean.

File: src/oic/oic/provider.py (hunk 1 above)
Comment (rohe) [reply src/oic/oic/provider.py]: OK

File: src/oic/oic/provider.py (hunk 2 above)
Comment (rohe) [reply src/oic/oic/provider.py]: OK

File: src/oic/oic/provider.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (tpazderka) [reply src/oic/oic/provider.py]: The ordering of imports is probably wrong. I think that in the current configuration, `future` imports should be on top.

**Comment by tpazderka** [comment]
There is still a conflict in oic/provider.py which is preventing the CI in running.

**Comment by tpazderka** [comment]
Closing in favor of #590

**Pull request closed without merging**